homeassistant/components/elkm1/__init__.py

```diff
@@ -187,7 +194,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     _LOGGER.debug("Setting up elkm1 %s", conf["host"])
 
-    config = {"temperature_unit": conf[CONF_TEMPERATURE_UNIT]}
+    temperature_unit = TEMP_FAHRENHEIT
+    if conf[CONF_TEMPERATURE_UNIT] in (BARE_TEMP_CELSIUS, TEMP_CELSIUS):
+        temperature_unit = TEMP_CELSIUS
```

Comment:
It's better to implement a config entry migration handler to update the config entry data to the correct state before the integration is set up.

See eg:
https://github.com/home-assistant/core/blob/d672eed3319a3a2e8073d5bc2702da47f1ee4d15/homeassistant/components/huawei_lte/__init__.py#L541-L551

Reply:
Thanks. Will look at cleaning this up once everything settles a bit more.